src/dirent.rs

```diff
@@ -0,0 +1,40 @@
+use {Result, Error, Errno};
+use libc::{self, DIR};
+use std::os::unix::io::RawFd;
+use errno;
+
+pub struct Dir {
+    pub handle: *mut DIR,
```

Comment:
I currently leave it private as I have implemented all functions, where the dir pointer is useful. If it turns out, that I missed a use case, it can be still added.